### PR TITLE
[WIP] Mobile Storage Creep

### PR DESCRIPTION
**Pull request by Spedwards**
Mobile Storage Creep. Currently a WIP.

Waiting for the viral tasks PR to get merged so I can add this task to the list.

TODO:

 - Delivering Action: A creep will request a resource, and the mobile storage will move to that creep and transfer to it. Ideally, taking the distance into account.
 - Storage Action: Similar to guarding action. Idles at the flag placed.
 - Add Mobile Storage as a valid target to reallocate energy.

### CODE REVIEW

File: global.js [hunk 1]
```diff
@@ -62,6 +62,10 @@ mod.FLAG_COLOR = {
         }
     },
     //COLOR_PURPLE,
+    economy: {
+        color: COLOR_PURPLE,
+        secondaryColor: COLOR_PURPLE,
+    },
```
Comment (karlthepagan) [inline global.js]: white/yellow flag exists and is suitable for this purpose

Comment (Spedwards) [inline global.js]: Yeah, the Purple/Purple was more of a placeholder for the time being.

File: task.mobileStorage.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            spawning: [],
+            running: [],
+        };
+    }
+    return flag.memory.tasks.mobileStorage;
+};
+
+mod.creep = {
+    fixedBody: [CARRY,CARRY,CARRY,CARRY,CARRY,CARRY,CARRY,CARRY,CARRY,CARRY,CARRY,CARRY,CARRY,CARRY,MOVE,MOVE,MOVE,MOVE,MOVE,MOVE,MOVE],
+    multiBody: [CARRY,CARRY,MOVE],
```
Comment (karlthepagan) [inline task.mobileStorage.js]: my previous thinking on creep design was 1 MOVE and the rest carry

however, this is highly dependent on the actions

**Comment by Perrytheplatypus43** [comment]
What will this be used for, will it be used as a storage for remote mining rooms?

**Comment by Spedwards** [comment]
@Perrytheplatypus43 Uses could be for anything. The task is mostly a basis for viral tasks. I'm sure we can come up with an idea for uses in public.

**Pull request closed without merging**